### PR TITLE
Implements installing a mackerel plugin by using registry

### PR DESCRIPTION
**Pull request by shibayu36**
If this pull request is merged, We can use `mkr plugin install <pluginName>@<releaseTag>` feature.

For example, mackerel-plugin-sample is defined at https://github.com/mackerelio/plugin-registry/blob/master/plugins/mackerel-plugin-sample.json , so we can install mackerel-plugin-sample by `mkr plugin install mackerel-plugin-sample@v0.0.1`.

## How to try this feature
Change commands.go like below, and execute `go run $(go list -f '{{join .GoFiles " "}}') plugin install --prefix ./mackerel-agent/plugins mackerel-plugin-sample@v0.0.1` in your terminal.

```diff
diff --git a/commands.go b/commands.go
index e2df850..d53212a 100644
--- a/commands.go
+++ b/commands.go
@@ -11,7 +11,7 @@ import (
        "github.com/Songmu/prompter"
        mkr "github.com/mackerelio/mackerel-client-go"
        "github.com/mackerelio/mkr/logger"
-       _ "github.com/mackerelio/mkr/plugin"
+       "github.com/mackerelio/mkr/plugin"
        "gopkg.in/urfave/cli.v1"
 )

@@ -47,7 +47,7 @@ var Commands = []cli.Command{
        commandAlerts,
        commandDashboards,
        commandAnnotations,
-       // plugin.CommandPlugin,
+       plugin.CommandPlugin,
 }
```

### CODE REVIEW

File: plugin/client.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package plugin
```
Comment (shibayu36) [inline plugin/client.go]: I separated http client function from plugin/commands.go

File: plugin/commands.go [hunk 2]
```diff
@@ -98,29 +99,20 @@ func setupPluginDir(pluginDir string) (string, error) {
 	return pluginDir, nil
 }
 
-// Download plugin artifact from `url` to `workdir`,
+// Download plugin artifact from `u`(URL) to `workdir`,
 // and returns downloaded filepath
-func downloadPluginArtifact(url, workdir string) (fpath string, err error) {
-	logger.Log("", fmt.Sprintf("Downloading %s", url))
+func downloadPluginArtifact(u, workdir string) (fpath string, err error) {
```
Comment (shibayu36) [inline plugin/commands.go]: I imported `url` package, so I changed variable name.

File: plugin/commands.go [hunk 3]
```diff
@@ -203,25 +198,77 @@ func newInstallTargetFromString(target string) (*installTarget, error) {
...
+		// TODO: Fetch latest release tag by github API
+		return "", fmt.Errorf("not implemented")
+	}
+
+	filename := fmt.Sprintf("%s_%s_%s.zip", url.PathEscape(repo), runtime.GOOS, runtime.GOARCH)
+	downloadURL := fmt.Sprintf(
+		"https://github.com/%s/%s/releases/download/%s/%s",
+		url.PathEscape(owner),
+		url.PathEscape(repo),
+		url.PathEscape(it.releaseTag),
```
Comment (shibayu36) [inline plugin/commands.go]: I had forgotten escaping user input for URL...

File: plugin/client.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// client provides utilities for http request
+type client struct {
+	userAgent string
```
Comment (astj) [inline plugin/client.go]: Are there any use cases for this? I think we can always use `defaultUserAgent`.

**Pull request merged**